### DISCUSSION TITLE
Convert AAC to another Audio Codec

### DISCUSSION THREAD

**Discussion by Taps7734** [bluenviron/mediamtx#2827] [Questions]
### Question

Is there a way to stream from OBS using an RTMP link and convert the audio to something besides AAC so that it will be used in WebRTC connections?

OBS uses either "FFmpeg AAC" or "CoreAudio AAC" (if installed).  

Is there a way to convert that to one of the WebRTC compliant codecs ( Opus, etc) for any WebRTC connection?

**Comment by AseCoder** [comment]
I had a similar question and I created a discussion when the recommended solution didn't work: #2787. Please try the steps listed there. I would be delighted to hear if you run into the same problem as I do.

**Comment by aler9** [comment]
Hello, OBS supports sending out audio directly in Opus, so no server-side conversion should be needed. Just do not use the RTMP protocol, but either RTSP or WebRTC. The README explains how to send out Opus with OBS and RTSP:

https://github.com/bluenviron/mediamtx#obs-studio

> If you want to generate a stream that can be read with WebRTC, open Settings -> Output -> Recording and use the following parameters:

>    FFmpeg output type: Output to URL
>    File path or URL: rtsp://localhost:8554/mystream
>    Container format: rtsp
>    Check show all codecs (even if potentically incompatible
>    Video encoder: h264_nvenc (libx264)
>    Video encoder settings (if any): bf=0
>    Audio track: 1
>    Audio encoder: libopus

>Then use the button Start Recording (instead of Start Streaming) to start streaming.

[a comment by an automated account is not shown]
